tests/test_mlkit.py

```diff
@@ -100,35 +102,94 @@ def test_get_model_validation_errors(self):
         #Empty model-id
         with pytest.raises(ValueError) as err:
             mlkit.get_model('')
-        self.check_error(err.value, ValueError, 'Model ID format is invalid.')
+        check_error(err.value, ValueError, 'Model ID format is invalid.')
 
         #None model-id
         with pytest.raises(TypeError) as err:
             mlkit.get_model(None)
-        self.check_error(err.value, TypeError, 'Model ID must be a string.')
+        check_error(err.value, TypeError, 'Model ID must be a string.')
 
         #Wrong type
         with pytest.raises(TypeError) as err:
             mlkit.get_model(12345)
-        self.check_error(err.value, TypeError, 'Model ID must be a string.')
+        check_error(err.value, TypeError, 'Model ID must be a string.')
 
         #Invalid characters
         with pytest.raises(ValueError) as err:
             mlkit.get_model('&_*#@:/?')
-        self.check_error(err.value, ValueError, 'Model ID format is invalid.')
+        check_error(err.value, ValueError, 'Model ID format is invalid.')
 
     def test_get_model_error(self):
-        _, recorder = self._instrument_mlkit_service(status=404, payload=_ERROR_RESPONSE)
+        recorder = instrument_mlkit_service(status=404, payload=_ERROR_RESPONSE)
         with pytest.raises(exceptions.NotFoundError) as err:
             mlkit.get_model(MODEL_ID_1)
-        self.check_firebase_error(err.value, ERROR_STATUS, ERROR_CODE, ERROR_MSG)
+        check_firebase_error(err.value, ERROR_STATUS, ERROR_CODE, ERROR_MSG)
         assert len(recorder) == 1
         assert recorder[0].method == 'GET'
-        assert recorder[0].url == self._get_url(PROJECT_ID, MODEL_ID_1)
+        assert recorder[0].url == self._url(PROJECT_ID, MODEL_ID_1)
 
     def test_no_project_id(self):
         def evaluate():
             app = firebase_admin.initialize_app(testutils.MockCredential(), name='no_project_id')
             with pytest.raises(ValueError):
                 mlkit.get_model(MODEL_ID_1, app)
         testutils.run_without_project_id(evaluate)
+
+class TestDeleteModel(object):
+    """Tests mlkit.delete_model."""
+    @classmethod
+    def setup_class(cls):
+        cred = testutils.MockCredential()
+        firebase_admin.initialize_app(cred, {'projectId': PROJECT_ID})
+
+    @classmethod
+    def teardown_class(cls):
+        testutils.cleanup_apps()
+
+    @staticmethod
+    def _url(project_id, model_id):
+        return BASE_URL + 'projects/{0}/models/{1}'.format(project_id, model_id)
+
+    def test_delete_model(self):
+        recorder = instrument_mlkit_service(status=200, payload=_EMPTY_RESPONSE)
+        mlkit.delete_model(MODEL_ID_1) # no response for delete
+        assert len(recorder) == 1
+        assert recorder[0].method == 'DELETE'
+        assert recorder[0].url == TestDeleteModel._url(PROJECT_ID, MODEL_ID_1)
+
+    def test_delete_model_validation_errors(self):
```

Comment:
Since this seems to come up repeatedly, how about using a parameterized test?

```
invalid_model_id_args = [
  (None, TypeError, 'Model ID must be a string'),
  (None, ValueError, 'Model ID format is invalid'), 
  # other test cases
]

@pytest.mark.parametrize('model_id, exc_type, error_message', invalid_model_id_args)
def delete_model_validation_errors(self, model_id, exc_type, error_message):
    with pytest.rases(exc_type) as err:
        mlkit.delete_mode(model_id)
    assert str(err.value) == error_message
```

You can do a similar test for `get_model()`.

Reply:
done